recipes/openai_textcat.py

```diff
@@ -0,0 +1,287 @@
+        return output
+
+    def _parse_binary(self, example: Dict, response: str) -> Dict:
+        """Parse binary TextCat where the 'answer' key means it's a positive class."""
+        output = self._parse_output(response)
+        example["answer"] = output["answer"].lower() == "accept"
+        example["meta"] = {
+            "answer": output["answer"].upper(),
+            "reason": output["reason"],
+        }
+        example["label"] = self.labels[0]
+        return example
+
+    def _parse_multi(self, example: Dict, response: str) -> Dict:
+        """Parse multilabel TextCat."""
+        output = self._parse_output(response)
+        example["options"] = [{"id": label, "text": label} for label in self.labels]
+        example["meta"]["reason"] = output["reason"]
+        example["answer"] = "accept"
+        # Filter removes any empty strings
+        example["accept"] = list(
+            filter(
+                None, [normalize_label(s.strip()) for s in output["answer"].split(",")]
+            )
+        )
+        return example
+
+
+@prodigy.recipe(
+    # fmt: off
+    "textcat.openai.correct",
+    input_path=("Path to jsonl data to annotate", "positional", None, Path),
+    prompt_path=("Path to jinja2 prompt template", "positional", None, Path),
+    labels=("Labels (comma delimited)", "option", "L", lambda s: s.split(",")),
+    lang=("Language to initialize spaCy model", "option", "l", str),
+    model=("GPT-3 model to use for completion", "option", "m", str),
+    batch_size=("Batch size to send to OpenAI API", "option", "b", int),
+    segment=("Split sentences", "flag", "S", bool),
+    examples_path=("Examples file to help define the task", "option", "e", Path),
+    max_examples=("Max examples to include in prompt", "option", "n", int),
+    exclusive_classes=("Make the classification task exclusive", "flag", "E", bool),
+    verbose=("Print extra information to terminal", "flag", "v", bool),
+    # fmt: on
+)
+def textcat_openai_correct(
+    dataset: str,
+    input_path: Path,
+    prompt_path: Path,
+    labels: List[str] = None,
+    lang: str = "en",
+    model: str = "text-davinci-003",
+    batch_size: int = 10,
+    segment: bool = False,
+    examples_path: Optional[Path] = None,
+    max_examples: int = 2,
+    exclusive_classes: bool = False,
+    verbose: bool = False,
+):
+    api_key, api_org = get_api_credentials(model)
+    examples = read_prompt_examples(examples_path, example_class=TextCatPromptExample)
+    if labels is None:
+        msg.fail("textcat.teach requires at least one --label", exits=1)
+    nlp = spacy.blank(lang)
+
+    if segment:
+        nlp.add_pipe("sentencizer")
+
+    # Create partial render of the template
+    if not exclusive_classes and len(labels) == 1:
+        msg.warn(
+            "Binary classification should always be exclusive. Setting "
+            "`exclusive_classes` parameter to True"
+        )
+        exclusive_classes = True
+
+    template = Template(
```

Comment:
Doesn't `load_template` always return a `Template` object already?

Reply:
The problem with this unfortunately is that I want to partially render the template 🤔 I don't want to pass `exclusive_classes` nor `labels` to the main suggester class. I'll think of a more elegant ways to do this (maybe via kwargs or something)